components/supervisor/pkg/ports/exposed-ports.go

```diff
@@ -174,14 +220,19 @@ func (g *GitpodExposedPorts) doExpose(req *exposePortRequest) {
 
 // Expose exposes a port to the internet. Upon successful execution any Observer will be updated.
 func (g *GitpodExposedPorts) Expose(ctx context.Context, local uint32, public bool) <-chan error {
-	v := "private"
-	if public {
-		v = "public"
+	if !public {
+		if !g.existInLocalExposed(local) {
+			g.localExposedPort = append(g.localExposedPort, local)
+			g.localExposedNotice <- struct{}{}
+		}
+		c := make(chan error)
+		close(c)
+		return c
```

Comment:
Should we remove from this list if we make a port public?

Reply:
We make ports public/private in `vscode` and `gp-cli` by calling server API, so we don't need to handle them in `supervisor`.

In supervisor methods below will call it
- Private method `autoExpose`
  - (internal) Will be called for according to config (`.gitpod.yml`)
  - (internal) Will be called after ports are served
  - (exposed) Will be called by `RetryAutoExpose` has no definition for `visibility`

https://github.com/gitpod-io/gitpod/blob/bca27705c6499b2319c1fbd2673b7dda285fa245/components/supervisor-api/port.proto#L77-L79
- (exposed) Method `ExposePort` has no definition for `visibility`too

https://github.com/gitpod-io/gitpod/blob/bca27705c6499b2319c1fbd2673b7dda285fa245/components/supervisor-api/control.proto#L22-L27